utils/s2n_random.c

```diff
@@ -34,6 +34,15 @@
  * - https://stackoverflow.com/a/5724485
  * - https://stackoverflow.com/a/5583764
  */
+#if defined(S2N_ADDRESS_SANITIZER)
+    /**
+     * Marks that a function should be ignored for address sanitizer jobs
+     */
+    #define ASAN_IGNORE __attribute__((no_address_safety_analysis))
+#else
+    #define ASAN_IGNORE
+#endif /* defined(S2N_ADDRESS_SANITIZER) */
```

Comment:
We should probably put this in a shared header so it can be reused, maybe s2n_utils/s2n_safety.h?